newsfragments/271.breaking.rst

```diff
@@ -0,0 +1 @@
+Updates to the latest ``eth-typing`` to use new ``ABI`` types for improved type checking in existing ``ABI`` utility functions.
```

Comment:
If this is considered breaking (I go back and forth on this... I don't think this would be breaking due to dependency resolution handling older support of the library), we should think about introducing an intermediate PR between this and the current state of things before this gets merged.

That PR would have the deprecation that exists here on `collapse_if_tuple()` and it wouldn't be typed as accurately if that method relies on the new *eth-typing* minor version. We can merge that in, release a current version update with the breaking changes, and then merge this in and release a new major version.

Again, that's only if we are considering it breaking. I'd love to hear from @kclowes and @pacrob on this too.

Reply:
Current release of `eth-utils` passes tests with latest `eth-typing==4.3.1` present so no problem there.

Given the scope of changes here, a new major feels appropriate, actually breaking or not. With that in mind, doing the separate current-major release to deprecate, then a new-major to drop `collapse_if_tuple` would be great. Nice to wrap it all up now.

Reply:
Here's the change to add that deprecation: https://github.com/ethereum/eth-utils/pull/279

Reply:
[Here](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api) is what semver has to say. Basically, you can update your own dependencies as long as it doesn't break the API. I do think that we have called this sort of dependency bump a breaking change though in the past, so I think it's a grey area. We had one complaint in eth-account somewhat recently that we added a dependency (pydantic) and it wasn't highlighted as a breaking change